.github/scripts/simple-smoke-test.sh

```diff
@@ -1,49 +1,88 @@
 #!/usr/bin/env bash
 
-set -e
-set -o pipefail
+set -o errexit -o nounset -o pipefail
 
 function test_docker_image() {
     local image=$1
     local container_name=$2
     local expected_distribution_type=$3
+    local expected_version=$4
 
-    if [ "$(docker ps --all --quiet --filter name="$container_name")" ]; then
-      echo "Removing existing '$container_name' container"
-      docker container rm --force "$container_name"
+    remove_container_if_exists "${container_name}"
+
+    check_distribution_type "${image}" "${expected_distribution_type}"
+    check_version "${image}" "${expected_version}"
+
+    test_map_read_write "${image}" "${container_name}"
+}
+
+function remove_container_if_exists() {
+    local container_name=$1
+
+    local containers
+    containers=$(docker ps --all --quiet --filter name="${container_name}")
+
+    if [[ -n ${containers} ]]; then
```

Comment:
any chance could be more than one running container?

Reply:
> any chance could be more than one running container?

Not sure what you mean?
This is existing functionality.

Reply:
True its current functionality but i was just thinking could the command return more than one container running.
But now I am thinking can two containers of the name be running at the same time?

Reply:
> True its current functionality but i was just thinking could the command return more than one container running. But now I am thinking can two containers of the name be running at the same time?

https://stackoverflow.com/questions/66591150/use-multiple-dockers-with-same-name#:~:text=You%20cannot%20have%20multiple%20containers,Docs%20regarding%20my%20below%20explanation.